### DISCUSSION TITLE
Changing theme programatically

### DISCUSSION THREAD

**Discussion by grizzlydevil** [fabiocaccamo/django-admin-interface#212] [Q&A] [answered]
Hello,
I was trying to change admin theme by setting theme.set_active(). When I enter my admin the old theme colours are still displayed, yet the new theme is set to active. After a couple of minutes the theme colour change.

Can you help me solve this?

**Comment by fabiocaccamo** [comment]
@grizzlydevil I think it's just browser cache (try to open the admin in a private window) because the app cache is deleted when calling `theme.set_active`.

**Reply by grizzlydevil** [reply]
I tried to open in different browser and in private window. Still the same.

**Reply by fabiocaccamo** [reply]
> I tried to open in different browser and in private window. Still the same.

What lib version are you using?

**Reply by grizzlydevil** [reply]
'0.18.2'

**Reply by fabiocaccamo** [reply]
Could you upgrade to the latest version and check again?
Remember to run `python manage.py migrate admin_interface`.

**Reply by grizzlydevil** [reply]
hmm.. sadly no, still doesn't work. It still changes but in a couple of minutes.

**Reply by fabiocaccamo** [reply]
Please check here if there is something useful: https://github.com/fabiocaccamo/django-admin-interface/issues/99

**Reply by grizzlydevil** [reply, marked as answer]
I found out what the problem was. The cache in the django settings was not set. I set the cache to dummy cache and now everything works fine. Themes are changing instantly.
```python
CACHES = {
    'default': {
        'BACKEND': 'django.core.cache.backends.dummy.DummyCache',
    }
}
```

**Reply by fabiocaccamo** [reply]
That's it! I'm glad that it's not a library bug, thank you for the feedback, maybe it will be useful to someone else.
